packages/driver/cypress/integration/e2e/multi-domain/multi_domain_uncaught_errors_spec.ts

```diff
@@ -0,0 +1,177 @@
+          expect(err.message).to.not.include('https://on.cypress.io/uncaught-exception-from-application')
+          // @ts-ignore
+          expect(err.docsUrl).to.deep.eq(['https://on.cypress.io/uncaught-exception-from-application'])
+
+          expect(uncaughtExceptionSpy.notCalled).to.be.true
+          expect(runnable === r).to.be.true
+        })
+
+        cy.switchToDomain('foobar.com', () => {
+          cy.get('.trigger-async-error').click()
+
+          // add the cy.wait here to keep commands streaming in,
+          // forcing the switchToDomain callback window to be open long enough for an async error to occur
+          cy.wait(1000)
+        })
+      })
+
+      it('passes the current test/command if async errors are thrown from the switchToDomain context, but the callback window is now closed', () => {
+        const uncaughtExceptionSpy = cy.spy()
+        const failureSpy = cy.spy()
+
+        cy.on('uncaught:exception', uncaughtExceptionSpy)
+
+        cy.on('fail', failureSpy)
+
+        cy.switchToDomain('foobar.com', () => {
+          // the async error here should be thrown AFTER the current command and test has finished, resulting in a passed test with no fail being triggered in the primary
+          cy.get('.trigger-async-error').click()
+        }).then(() => {
+          expect(uncaughtExceptionSpy.notCalled).to.be.true
+          expect(failureSpy.notCalled).to.be.true
+        })
+      })
+    })
+
+    describe('unhandled rejections', () => {
+      it('unhandled rejection triggers uncaught:exception and has promise as third argument', (done) => {
+        cy.switchToDomain('foobar.com', done, () => {
+          const r = cy.state('runnable')
+
+          cy.once('uncaught:exception', (err, runnable, promise) => {
+            expect(err.stack).to.include('promise rejection')
+            expect(err.stack).to.include('one')
+            expect(err.stack).to.include('two')
+            expect(err.stack).to.include('three')
+            expect(runnable === r).to.be.true
+            expect(promise).to.be.a('promise')
+
+            done()
+
+            return false
+          })
+
+          cy.get('.trigger-unhandled-rejection').click()
+        })
+      })
+
+      it('original error is not mutated for "unhandledrejection"', () => {
+        cy.switchToDomain('foobar.com', () => {
+          cy.once('uncaught:exception', () => false)
+
+          cy.get('.trigger-unhandled-rejection').click()
+          cy.get('.error-one').invoke('text').should('equal', 'promise rejection')
+          cy.get('.error-two').invoke('text').should('equal', 'promise rejection')
+        })
+      })
+    })
+
+    it('does not fail if thrown custom error with readonly name', (done) => {
+      cy.once('fail', (err) => {
+        // TODO: we cannot currently serialize custom user errors back to the primary
+        // ideally, we want the name of this error to be 'CustomError'
```

Comment:
We can also merge the name property into the deserialized error so that custom properties come through, but figured it might be a bit out of scope here.